vulture/config.py

```diff
@@ -0,0 +1,194 @@
+"""
+This module handles retrieval of configuration values from either the
+command-line arguments or the pyproject.toml file.
+"""
+from __future__ import print_function
+import argparse
+import sys
+from os.path import abspath, exists
+
+import toml
+
+from .version import __version__
+
+#: Possible configuration options and their respective defaults
+DEFAULTS = {
+    "min_confidence": 0,
+    "paths": [],
+    "exclude": [],
+    "ignore_decorators": [],
+    "ignore_names": [],
+    "make_whitelist": False,
+    "sort_by_size": False,
+    "verbose": False,
+}
+
+#: sentinel value to distinguish between "False" and "no default given"
+NO_DEFAULT = object()
+
+
+def from_dict(data):
+    """
+    Create a new config dictionary from an existing one, assign possible
+    defaults and warn about unprocessed options.
+    """
+    # keep a copy of the keys, so we can keep track of any unprocessed
+    # values.
+    remaining_keys = set(data.keys())
+
+    output = {}
+    for key, default in DEFAULTS.items():
+        output[key] = data.get(key, default)
+        remaining_keys.discard(key)
+    for remainder in sorted(remaining_keys):
+        print(
+            u"Unprocessed config option {}".format(remainder), file=sys.stderr
+        )
+    if remaining_keys:
+        sys.exit(u"Invalid config!")
```

Comment:
I think it would be cleaner to rename remaining_keys to unknown_keys and compute them with `unknown_keys = set(data) - set(DEFAULTS)` and then

    if unknown_keys:
        sys.exit("Unknown configuration keys: {}".format(sorted(unknown_keys)))

Reply:
sys.exit does not seem to write onto `stderr` (at least the test with `capsys` fails). So I kept the prints in there. But otherwise I implemented your suggestion in 3e21554

Reply:
Sounds good to me.